### DISCUSSION TITLE
Optimized parameters for Deepgram live streaming transcription (self-hosted)

### DISCUSSION THREAD

**Discussion by devalbham** [deepgram/community#998] [General help] [answered]
Hi, may I know what would be the optimized parameters for for Deepgram live streaming transcription.

I am using Websockets from my Frontend (react) to Backend (FastAPI) and another Websocket from my BE to Deepgram API (self-hosted)

Currently I am using. It's working ok, but was thinking if I can improve it further. 

sampleRate = 16000
channelCount = 2
chunkDuration = 200 //milliseconds

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Comment by devalbham** [comment]
Using self-hosted, live streaming Speech-to-text API
Request ID: 596640d9-14db-42b2-a52d-5fff7bd241a9

**Comment by deepgram-community** [comment, marked as answer]
It seems like you're using Deepgram's live streaming transcription with a self-hosted setup. The parameters you're currently using (sampleRate = 16000, channelCount = 2, chunkDuration = 200 milliseconds) are generally a good starting point. However, the optimal parameters can depend on various factors such as the quality of your audio input, the network conditions, and the specific requirements of your application.

If you're looking to improve the performance, you might consider the following:

1. **Audio Quality**: Ensure that the audio input is clear and free of background noise. The quality of the transcription is directly related to the quality of the audio input.

2. **Network Conditions**: If your network conditions are not stable, you might need to adjust the chunkDuration. A smaller chunkDuration means more frequent but smaller data transmissions, which might be more robust under unstable network conditions.

3. **Application Requirements**: Depending on your application, you might need to prioritize different aspects. For example, if real-time performance is crucial, you might need to use a smaller chunkDuration to get transcription results faster. On the other hand, if accuracy is more important, a larger chunkDuration might be better as it allows more context for the speech recognition.

Remember to test different parameters under realistic conditions to find the optimal settings for your specific use case. If you need further assistance, feel free to provide more details about your application and the issues you're facing.
> John Vajda
